core/src/androidTest/java/novoda/lib/sqliteprovider/ContentProviderTest.java

```diff
@@ -18,4 +28,46 @@ public void testMapping() throws Exception {
         assertTrue(cursor.getColumnIndexOrThrow("childs__id") > -1);
         assertTrue(cursor.getColumnIndexOrThrow("parents_name") > -1);
     }
+
+    public void test_GivenATableWithData_When_UpdatingNonexistentRow_Then_NumberOfAffectedRowsShouldBeZero() throws Exception {
+        givenATableWithData();
+
+        ContentValues values = new ContentValues(1);
+        values.put(COLUMN2, 1);
+
+        int numRows = new ExtendedSQLiteOpenHelper(getContext())
+                .getWritableDatabase()
+                .update(TABLE_NAME, values, COLUMN_PRIMARY_KEY + "=?", new String[]{"2"});
+
+        assertEquals(0, numRows);
+
+        numRows = getContext().getContentResolver().update(
+                Uri.parse("content://novoda.lib.sqliteprovider.test/" + TABLE_NAME),
+                values,
+                COLUMN_PRIMARY_KEY + "=?",
+                new String[]{"2"});
+
+        assertEquals(0, numRows);
+    }
+
+    private void givenATableWithData() throws IOException {
+        ExtendedSQLiteOpenHelper helper = new ExtendedSQLiteOpenHelper(getContext());
+        SQLiteDatabase db = helper.getWritableDatabase();
+        db.execSQL("CREATE TABLE IF NOT EXISTS `" + TABLE_NAME + "` (" +
+                           "`" + COLUMN_PRIMARY_KEY + "` INTEGER PRIMARY KEY," +
+                           "`" + COLUMN2 + "` INTEGER UNIQUE)");
+
+        ContentValues values = new ContentValues(2);
+        values.put(COLUMN_PRIMARY_KEY, 1);
+        values.put(COLUMN2, 1);
+        helper.getWritableDatabase().insert(TABLE_NAME, null, values);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        super.tearDown();
```

Comment:
nitpic: you don't actually have to call this, but what I was originally going to say was LIFO call this last line of hte method :)